launcher/internal/controller/bcsconfig_controller.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	bcsv1 "bcs.pod.launcher.intel/api/v1"
+	"bcs.pod.launcher.intel/resources_library/utils"
+)
+
+// BcsConfigReconciler reconciles a BcsConfig object
+type BcsConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	logr.Logger
+}
+
+// Information about rbac
+// groups=bcs.bcs.intel,resources=bcsconfigs,verbs=get;list;watch;create;update;patch;delete
+// groups=bcs.bcs.intel,resources=bcsconfigs/status,verbs=get;update;patch
+// groups=bcs.bcs.intel,resources=bcsconfigs/finalizers,verbs=update
+// groups=apps,resources=daemonsets;deployments,verbs=get;list;watch;create;update
+// groups="",resources=services;configmaps;persistentvolumes;persistentvolumeclaims,verbs=get;list;watch;create;update
+
+func (r *BcsConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	// MCM seamless start up
+	createResourceIfNotExists := func(resource client.Object, namespacedName types.NamespacedName) error {
+		err := r.Get(ctx, namespacedName, resource)
+		if err != nil {
```

Comment:
Does this `err` value only says the resource is not found? If there could be other reasons for an error, you probably need to return with an error here instead of creating the new resource.

Reply:
I mean something like `if errors.IsNotFound(err) {`

Reply:
The idea behind is to check if resource exists or not. If not, it creates one. I'd leave it in this form. In task for mcm, it can be more checked. It is just simple form of it